core/Tracker/PageUrl.php

```diff
@@ -44,6 +44,11 @@ public static function excludeQueryParametersFromUrl($originalUrl, $idSite)
         $parsedUrl = self::cleanupHostAndHashTag($parsedUrl, $idSite);
         $parametersToExclude = self::getQueryParametersToExclude($idSite);
 
+        if (!empty($parsedUrl['host']) && strpos($parsedUrl['host'], 'instagram.com') !== false) {
```

Comment:
@sgiehl couldn't find how this method is used in referrer url and also set a debug breakpoint when executing eg `matomo.php?rec=1&idsite=1&action_name=foo&send_image=1&urlref=https%3A%2F%2Fl.instagram.com%2F%3Fu%3Dhttps%253A%252F%252Fexample.com%252Fexample.com%26amp%3Be%3DBTPcuqWixl6Mf5hgYPp6wXIlstuaEdJssdYEvT9s8-6yme_lb275lY2Bwc-YvE-fZNtSKux4QB-v8xNk%26amp%3Bs%3D1` but it wouldn't go in there for the referrer url? (it's a new visit).

Is this executed for the referrer url?

Reply:
Sorry, my fault. Seems I didn't think that through. Thought that would be similar as the gclid parameter, but for sure that isn't the case. Will update the implementation.